spec/cheetah_spec.rb

```diff
@@ -757,6 +757,33 @@ def create_command(source, name: "command")
           expect(Cheetah.run("/bin/false", stdout: :capture, allowed_exitstatus: 1)).to eq(["", 1])
         end
       end
+
+      describe "changing environment" do
+        let(:command) do
+          create_command(<<-EOT)
+            echo -n ${CHEETAH_TEST?"NOT SET"}
+          EOT
+        end
+
+        it "runs command with given env" do
+          expect(
+            Cheetah.run(command, stdout: :capture, env: { "CHEETAH_TEST" => "OK" })
+          ).to eq "OK"
+        end
+
+        it "do not change env outside call" do
+          ENV["CHEETAH_TEST"] = "OK"
+          Cheetah.run(command, stdout: :capture, env: { "CHEETAH_TEST" => "fail" })
+          expect(ENV["CHEETAH_TEST"]).to eq "OK"
+        end
+
+        it "key with nil unsets value" do
+          ENV["CHEETAH_TEST"] = "OK"
+          # different shells use different exit value, so allow all of them
```

Comment:
That's because you use the wrong shell construct: `${VAR?error-message-if-unset}`. Use `${VAR-value-if-unset}` instead. See [Shell Parameter Expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html).